projects/backend/controllers/render.ts

```diff
@@ -305,20 +336,25 @@ export const renderItemController = async (req: Request, res: Response) => {
     }
 
     // find the corresponding furniture for the article so we can apply the fronts overrides
-    const idFurniturePair = front.collections
+    const groupIdToFurnitureAndTheme = front.collections
         .map(collection =>
-            collection.items.map((item): [number, PublishedFurniture] => [
-                item.internalPageCode,
-                item.furniture,
-            ]),
+            collection.items.map((item): [
+                number,
+                PublishedFurniture,
+                Theme,
+            ] => {
+                const mappedTheme = mapSwatchToTheme(front.swatch)
+
+                return [item.internalPageCode, item.furniture, mappedTheme]
+            }),
         )
         .reduce((acc, val) => acc.concat(val), [])
         .map(r => {
-            return { internalPageCode: r[0], furniture: r[1] }
+            return { internalPageCode: r[0], furniture: r[1], theme: r[2] }
```

Comment:
Again, not 100% on why these are array indexes or what we're using `reduce` for - maybe we can go through and see if there's a simpler way of doing the same thing?

Reply:
Agreed. Have taken a quick look and looks like this code can be refactored with some flatMaps etc. Issue is that we are currently set to compile to ES2017 (which doesn't have flat, flatMap etc). I wonder if its worth refactoring this is a separate PR?

Reply:
Yeah I think maybe just leave it for now if it was working ok and we're not really touching it with any changes?

Reply:
Feels a bit like opening a can of worms and there's plenty of other stuff to do.